### DISCUSSION TITLE
Color Picker Refresh/Revert

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#2857] [General] [answered]
Hello! Working on some editor inputs, and ran into this input style that includes a refresh/revert back option.

<img width="332" alt="Screen Shot 2022-12-02 at 2 11 36 PM" src="https://user-images.githubusercontent.com/87141551/205386774-f682ce88-84de-40a5-928c-4dfb40e8342b.png">

Wondering what your rec would be for adding in a refresh/revert icon - would the mockups below work? thank you!

<img width="350" alt="Screen Shot 2022-12-02 at 2 16 14 PM" src="https://user-images.githubusercontent.com/87141551/205387492-9459d271-c602-4188-b057-dfd7caa74268.png">

<img width="585" alt="Screen Shot 2022-12-02 at 2 17 54 PM" src="https://user-images.githubusercontent.com/87141551/205387775-64b5503c-e55e-4ef1-96c1-0098c62dd6d6.png">

<img width="636" alt="Screen Shot 2022-12-02 at 2 21 55 PM" src="https://user-images.githubusercontent.com/87141551/205388435-00a2459f-370e-4d0d-9f22-31bb5b604b96.png">

**Comment by serifluous** [comment, marked as answer]
Hey @ceyerkes !

A few points here:

1. We'll be adding a "Reset" icon to Paste soon! You can take a look at @brannytempy's branch in Figma here: https://www.figma.com/file/1Tz1QElYz96z20j81695tk/feat%3A-Adding-Segment-Icons?node-id=20745%3A146623&t=ieUftQlhJxhhF1H5-4 Let us know if you want us to get it in soon so you can start using it.

2. Unfortunately neither the "Reset" icon nor the color swatch circle can be placed in the field like you have mocked up, where they're not prefixes/suffixes on the form field and don't have the gray background behind them.

I put together some options for placement of both the icon and color swatch here, roughly based on #2786: https://www.figma.com/file/SpeLpBoqN2vZv8KWcSn0J1/Paste-Feedback?node-id=2110%3A412289&t=KNf8Fs8pkfllpXsG-0

![Screen Shot 2022-12-05 at 5 08 50 PM](https://user-images.githubusercontent.com/3821840/205781357-565de634-2a1f-419b-8a69-acc7557a330d.png)

Option 1 sacrifices some clarity about how to open the color picker for fewer elements on the screen. Option 2 gives a redundant way to open the color picker, but it could be clearer than just showing the swatch itself.

I'm still waiting on some input from the rest of the UX Infra team to see if there's a solution that uses [Combobox Primitive](https://paste.twilio.design/primitives/combobox-primitive) that gets closer to what you've designed, but this might be the workaround for now until we build out a true Color Picker component.

**Reply by ceyerkes** [reply]
Hey @serifluous! Thanks for laying out those 2 options. I think option #1 looks great - let's discuss more when we chat today. I'm definitely interested in using the reset icon soon for the color picker, as well as the undo/redo icons I need for the content editor.
